src/test/run-pass-fulldeps/derive-no-std.rs

```diff
@@ -8,12 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![feature(rand, collections, rustc_private)]
+#![feature(libc, lang_items, start)]
 #![no_std]
 
-extern crate rand;
-extern crate serialize as rustc_serialize;
-extern crate collections;
+extern crate libc; // to pull in libSystem on OSX
+
+#[lang = "panic_fmt"] fn panic_fmt() -> ! { loop {} }
+#[lang = "eh_personality"] extern fn eh_personality() {}
```

Comment:
Ah sorry I just mean that the point of this test is to exercise `#[derive]` in a `#![no_std]` crate. Having a no_std binary involves hokey unstable things like these lang items, so it may be best to have a library crate as `#![no_std]` (in the `src/test/auxiliary` folder) and then the binary here can just link to it with stable Rust.

That should exercise all the same functionality without the use of unstable code.